hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/ScheduleServiceImpl.java

```diff
@@ -54,40 +53,45 @@ public void registerSchemas(@NonNull final SchemaRegistry registry) {
 
     private Schema scheduleSchema() {
         // Everything in memory for now
-        return new Schema(CURRENT_VERSION) {
-            @SuppressWarnings("rawtypes")
-            @NonNull
-            @Override
-            public Set<StateDefinition> statesToCreate() {
-                final Codec<MerkleScheduledTransactionsState> migrationCodec =
-                        MonoMapCodecAdapter.codecForSelfSerializable(
-                                MerkleScheduledTransactionsState.CURRENT_VERSION,
-                                MerkleScheduledTransactionsState::new);
-                return Set.of(
-                        StateDefinition.singleton(SCHEDULING_STATE_KEY, migrationCodec),
-                        schedulesByIdDef(),
-                        schedulesByExpirySec(),
-                        schedulesByEquality());
-            }
-
-            @Override
-            public void migrate(@NonNull MigrationContext ctx) {
-                // Prepopulate the ScheduleServices state with default values for genesis
-                final var s = ctx.newStates().getSingleton(SCHEDULING_STATE_KEY);
-                s.put(new MerkleScheduledTransactionsState()); // never scheduled
-            }
-        };
+        return new ScheduleServiceSchema();
     }
 
-    private StateDefinition<ScheduleID, Schedule> schedulesByIdDef() {
-        return StateDefinition.inMemory(SCHEDULES_BY_ID_KEY, ScheduleID.PROTOBUF, Schedule.PROTOBUF);
-    }
+    private static final class ScheduleServiceSchema extends Schema {
+        public ScheduleServiceSchema() {
+            super(ScheduleServiceImpl.CURRENT_VERSION);
+        }
 
-    private StateDefinition<Long, List<Schedule>> schedulesByExpirySec() {
-        return StateDefinition.inMemory(SCHEDULES_BY_EXPIRY_SEC_KEY, LongCodec.SINGLETON, LIST_OF_SCHEDULE_CODEC);
-    }
+        @SuppressWarnings("rawtypes")
+        @NonNull
+        @Override
+        public Set<StateDefinition> statesToCreate() {
+            final Codec<MerkleScheduledTransactionsState> migrationCodec = MonoMapCodecAdapter.codecForSelfSerializable(
```

Comment:
Why do we still need this mono migration codec?

Reply:
I don't think we do, but it's not the subject of this PR.
I believe I have it removed in the next PR for schedule service.